indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -317,6 +322,157 @@ public void handle()
     }
   }
 
+  // same as submit supervisor logic
+  private class DynamicAllocationTasksNotice implements Notice
+  {
+    @Override
+    public void handle()
+    {
+      lock.lock();
+      try {
+        long nowTime = System.currentTimeMillis();
+        long minTriggerDynamicFrequency = Long.parseLong(String.valueOf(dynamicAllocationTasksProperties.getOrDefault("minTriggerDynamicFrequencyMillis", 1200000)));
+        // Only queue is full and over minTriggerDynamicFrequency can trigger scale out/in
+        // max(minTriggerDynamicFrequency, metricsCollectionRangeMillis)
+        if (spec.isSuspended()) {
+          log.info("[%s] supervisor is suspended, skip to check dynamic allocate task logic", dataSource);
+          return;
+        }
+        log.info("PendingCompletionTaskGroups is : " + pendingCompletionTaskGroups);
+        for (CopyOnWriteArrayList list : pendingCompletionTaskGroups.values()) {
+          if (!list.isEmpty()) {
+            log.info("Still hand off tasks unfinished, skip to do scale action [" + pendingCompletionTaskGroups + "]");
+            return;
+          }
+        }
+        if (nowTime - dynamicTriggerLastRunTime < minTriggerDynamicFrequency) {
+          log.info("NowTime - dynamicTriggerLastRunTime is [" + (nowTime - dynamicTriggerLastRunTime) + "]. Defined minTriggerDynamicFrequency is [" + minTriggerDynamicFrequency + "] , CLAM DOWN NOW !");
+          return;
+        }
+        if (!queue.isAtFullCapacity()) {
+          log.info("Metrics collection is not at full capacity, skip to check dynamic allocate task : [" + queue.size() + " vs " + queue.maxSize() + "]");
+          return;
+        }
+        List<Long> lags = collectTotalLags();
+        boolean allocationSuccess = dynamicAllocate(lags);
+        if (allocationSuccess) {
+          dynamicTriggerLastRunTime = nowTime;
+          queue.clear();
+        }
+      }
+      catch (Exception e) {
+        log.error(e, "Error, when parse DynamicAllocationTasksNotice");
+      }
+      finally {
+        lock.unlock();
+      }
+    }
+  }
+
+  private boolean dynamicAllocate(List<Long> lags) throws InterruptedException, ExecutionException, TimeoutException
+  {
+    // if supervisor is not suspended, ensure required tasks are running
+    // if suspended, ensure tasks have been requested to gracefully stop
+    log.info("[%s] supervisor is running, start to check dynamic allocate task logic", dataSource);
+    long scaleOutThreshold = Long.parseLong(String.valueOf(dynamicAllocationTasksProperties.getOrDefault("scaleOutThreshold", 5000000)));
+    long scaleInThreshold = Long.parseLong(String.valueOf(dynamicAllocationTasksProperties.getOrDefault("scaleInThreshold", 1000000)));
+    double triggerSaleOutThresholdFrequency = Double.parseDouble(String.valueOf(dynamicAllocationTasksProperties.getOrDefault("triggerSaleOutThresholdFrequency", 0.3)));
```

Comment:
I think these may be spelling mistakes in variable name and config value for this and next config. `triggerSale*` --> `triggerScale*` ?

Reply:
Done.